lib/Controller/WopiController.php

```diff
@@ -595,7 +608,22 @@ public function putRelativeFile($fileId,
 			return new JSONResponse([], Http::STATUS_FORBIDDEN);
 		}
 
-		list($fileId, ,) = Helper::parseFileId($fileId);
+		switch ($wopiOverride) {
+			case 'LOCK':
+				return $this->lock($wopi, $wopiLock);
+			case 'UNLOCK':
+				return $this->unlock($wopi, $wopiLock);
+			case 'REFRESH_LOCK':
+				return $this->refreshLock($wopi, $wopiLock);
+			case 'GET_LOCK':
+				return $this->getLock($wopi, $wopiLock);
+			case 'RENAME_FILE':
+				break; //FIXME: Move to function
+			default:
+				break; //FIXME: Move to function and add error for unsupported method
+		}
+
+
 		$isRenameFile = ($this->request->getHeader('X-WOPI-Override') === 'RENAME_FILE');
 
 		if (!$wopi->getCanwrite()) {
```

Comment:
Can a user without write permission lock the file?

Reply:
Resolving this because I understood the code a bit better from your other comment. I am assuming the rights handling should be done by each operation, In this case its handled by the lock manager